pytest/tests/loadtest/locust/common/congestion.py

```diff
@@ -66,6 +66,7 @@ def sender_account(self) -> base.Account:
 
 @events.init.add_listener
 def on_locust_init(environment, **kwargs):
+    base.INIT_DONE.wait()
```

Comment:
I wonder if unfortunate scheduling could lead to a deadlock. I'm not sure how locust schedules these events internally or how it may change with new versions.

To avoid starring at a blank terminal in case the deadlock happens, perhaps it would make sense to add a defensive check like this?
```python
# I think that's how one would add a timeout and check if it was triggered
init_done = base.INIT_DONE.wait(timeout=300)
if not init_done:
  raise SystemExit("base initialization took longer than 5 minutes")
```

But I'm not sure if we can add a safe timeout value. So really, I'm no sure if it is worth adding. Feel free to merge as-is without my suggestion.

Reply:
yeah noted. I guess it's kind of tough to think of a safe timeout value as you said, so I'll just merge it for now, although a deadlock would be kind of sad indeed... But trying to think about how that would happen, it seems like it should be unlikely (famous last words), because we have a bunch of init functions all waiting on one single init function to complete, with no cycle anywhere. So unless for some reason the `on_locust_init` function in `base.py` hangs (which would indicate an unrelated problem), we should be good I think?

Reply:
I was mostly worried that the one crucial event which all others are waiting for gets scheduled in the same thread as another event, but afterwards. Again, I'm not sure how scheduling works here but if say locust has synchronous code that executes these events one by one and waits for completion of event 1 before starting event 2, this could easily lead to a deadlock.

Anyway, this is not production code. And the type of deadlock I'm worried about should either happen almost always or not at all, so I'm not too worried about it.